app/models/shelter.rb

```diff
@@ -17,6 +17,7 @@ def average_review
   end
 
   def number_applications
-    ApplicationPet.joins('JOIN pets on application_pets.pet_id = pets.id').joins('JOIN shelters on shelters.id = pets.shelter_id').where("shelters.id = #{id}").count
+      pets.joins(:applications).select('applications.*').distinct.count
```

Comment:
faster and more concise way of accessing the number of `applications` per `shelter`